webui/src/lib/components/repository/tree.jsx

```diff
@@ -569,52 +571,66 @@ export const URINavigator = ({
-            }}
-          >
-            {reference.type === RefTypeCommit
-              ? reference.id.substr(0, 12)
-              : reference.id}
-          </Link>
-          <strong>{"/"}</strong>
-        </>
-      ) : (
-        <>
-          <Link href={pathURLBuilder(params, { path: "" })}>{relativeTo}</Link>
-          <strong>{"/"}</strong>
-        </>
-      )}
-
-      {parts.map((part, i) => {
-        const path =
-          parts
-            .slice(0, i + 1)
-            .map((p) => p.name)
-            .join("/") + "/";
-        const query = { path, ref: reference.id };
-        const edgeElement =
-          isPathToFile && i === parts.length - 1 ? (
-            <span>{part.name}</span>
-          ) : (
-            <>
-              <Link href={pathURLBuilder(params, query)}>{part.name}</Link>
-              <strong>{"/"}</strong>
-            </>
-          );
-        return <span key={part.name}>{edgeElement}</span>;
-      })}
+    <span className="lakefs-uri" style={{ display: "flex" }}>
+      <div>
+        {relativeTo === "" ? (
+          <>
+            <strong>{"lakefs://"}</strong>
+            <Link href={{ pathname: "/repositories/:repoId/objects", params }}>
+              {repo.id}
+            </Link>
+            <strong>{"/"}</strong>
+            <Link
+              href={{
+                pathname: "/repositories/:repoId/objects",
+                params,
+                query: { ref: reference.id },
+              }}
+            >
+              {reference.type === RefTypeCommit
+                ? reference.id.substr(0, 12)
+                : reference.id}
+            </Link>
+            <strong>{"/"}</strong>
+          </>
+        ) : (
+          <>
+            <Link href={pathURLBuilder(params, { path: "" })}>{relativeTo}</Link>
+            <strong>{"/"}</strong>
+          </>
+        )}
+
+        {parts.map((part, i) => {
+          const path =
+            parts
+              .slice(0, i + 1)
+              .map((p) => p.name)
+              .join("/") + "/";
+          const query = { path, ref: reference.id };
+          const edgeElement =
+            isPathToFile && i === parts.length - 1 ? (
+              <span>{part.name}</span>
+            ) : (
+              <>
+                <Link href={pathURLBuilder(params, query)}>{part.name}</Link>
+                <strong>{"/"}</strong>
+              </>
+            );
+          return <span key={part.name}>{edgeElement}</span>;
+        })}
+      </div>
+      <ButtonGroup className="commit-actions" style={{marginLeft: 'auto' }}>
+        <ClipboardButton
+            text={`lakefs://${repo.id}/${reference.id}/${path}`}
+            variant="outline-primary"
+            size="sm"
+            onSuccess={noop}
+            onError={noop}
+            className={"me-1"}
+            tooltip={"copy URI to clipboard"}
+            hidden={isPathToFile}
+            disabled={isPathToFile}/>
```

Comment:
Why disable it when it's not a file? I think copying prefixes is valuable too.

Reply:
We're disabling when it is a file - since the object viewer has already a copy and download buttons